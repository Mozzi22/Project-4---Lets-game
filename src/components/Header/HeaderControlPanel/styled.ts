import styled from 'styled-components';

export const StControl = styled.div`
        display: flex;
        justify-content: flex-end;
        align-items: center;
        height: 100%;
        width: 80%;
        margin-right: 20px;
        p { color: #fff }
`;
