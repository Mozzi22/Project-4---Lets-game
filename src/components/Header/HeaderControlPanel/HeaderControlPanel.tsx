import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import Button from 'src/components/UI/Button';
import Select from 'src/components/UI/Select';
import { Theme } from 'src/components/Hocs/withTheme';
import { StOption } from 'src/components/UI/Select/styled';
import { StControl } from './styled';

interface IHeaderControlPanel {
    themeMode: string;
}

const options = [
    { id: 1, label: 'en', value: 'en' },
    { id: 2, label: 'ru', value: 'ru' },
];

const HeaderControlPanel = ({
}: IHeaderControlPanel) => {
    const { t, i18n } = useTranslation();
    const { theme, changeTheme } = useContext(Theme);
    const [selected, setSelected] = React.useState('');

    React.useEffect(() => {
        const lastSelected = localStorage.getItem('lang');
        setSelected(lastSelected);
    }, [localStorage.getItem('lang')]);

    const onChangeLanguage = (lng, i18n) => {
        i18n.changeLanguage(lng);
        localStorage.setItem('lang', lng);
    };

    const handleChangeLanguage = (e) => {
        onChangeLanguage(e.target.value, i18n);
    };

    const toggleThemeMode = () => changeTheme(theme === 'dark' ? 'light' : 'dark');

    return (
        <StControl>
            <Button
                fontSize='26px'
                width='40px'
                height="40px"
                margin='0 15px 0'
                borderRadius='0px'
                onClick={toggleThemeMode}
                bgColor='transparent'
                backgroundSize='cover'
            />
            <Select
                id={options.id}
                options={options}
                onChange={handleChangeLanguage}
                value={selected}
                width="70px"
                bgColor=""
            >
            {options.map(el => (
                <StOption
                    key={el.id}
                    id={el.id}
                    value={el.value}
                >
                    {el.label}
                </StOption>
                ))
            }
            </Select>
        </StControl>
    );
};

export default HeaderControlPanel;
