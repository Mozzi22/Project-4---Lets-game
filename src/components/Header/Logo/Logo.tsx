import React from 'react';
import { useTranslation } from 'react-i18next';
import { useTheme } from 'src/components/Hooks/useTheme';
import Button from 'src/components/UI/Button';
import { StLogo } from './styled';

const Logo = ({ history }) => {
    const { t } = useTranslation();
    const { colors, theme } = useTheme();

    const goToMain = () => {
        const { pathname } = history.location;
        history.push('/main');
    };
    const goToStats = () => {
        const { pathname } = history.location;
        history.push('/statistics');
    };

    return (
        <StLogo
            theme={theme}
            colors={colors}
        >
            <img src="/public/assets/images/game-logo.svg" onClick={goToMain} />
            <Button
                content={t('main_page')}
                width="110px"
                fontSize='18px'
                backgroundSize="0"
                padding="0"
                margin="0"
                bgColor="transparent"
                onClick={goToMain}
            />
                <Button
                    content={t('statistics')}
                    width="110px"
                    fontSize='18px'
                    backgroundSize="0"
                    padding="0"
                    margin="0"
                    bgColor="transparent"
                    onClick={goToStats}
                />
        </StLogo>
    );
};

export default Logo;
