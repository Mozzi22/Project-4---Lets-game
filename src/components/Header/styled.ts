import styled from 'styled-components';
import { IAppDiv } from 'src/components/App/styled';
import { colors } from 'src/components/UI/themeStyles';

export const StHeader = styled.div < IAppDiv > `
    border-bottom: 1px solid;
    border-color: ${({ colors, theme }) => colors[theme].borderColor};
    display:flex;
    justify-content: flex-start;
    align-items: center;
    height:80px;
    ${({ transition = 'all 300ms ease-in-out' }) => transition && `transition: ${transition}`};
`;
