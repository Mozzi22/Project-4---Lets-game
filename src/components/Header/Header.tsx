import React from 'react';
import { useTheme } from 'src/components/Hooks/useTheme';
import { StHeader } from './styled';
import Nav from './Nav';
import HeaderControlPanel from './HeaderControlPanel';

const Header = () => {
    const { colors, theme } = useTheme();
    return (
        <StHeader
            theme={theme}
            colors={colors}
        >
            <Nav />
            <HeaderControlPanel />
        </StHeader >
    );
};

export default Header;
