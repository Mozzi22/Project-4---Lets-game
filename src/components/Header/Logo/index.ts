import { withRouter } from 'react-router';
import { connect } from 'react-redux';
import { setValue } from 'src/Store/user/actions';
import Logo from './Logo';

const mapDispatchToProps = dispatch => ({ setValue: payload => dispatch(setValue(payload)) });

export default connect(null, mapDispatchToProps)(withRouter(Logo));
