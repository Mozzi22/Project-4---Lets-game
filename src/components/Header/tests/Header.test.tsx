import React from 'react';
import { shallow, mount } from 'enzyme';
import Header from '../Header';
import HeaderControlPanel from '../HeaderControlPanel';
import Logo from '../Logo';

describe('Header', () => {
  it('Should match snapshot', () => {
    const component = shallow(<Header />);
    expect(component.html()).toMatchSnapshot();
  });
    it('Should have Logo', () => {
    const component = mount(<Logo />);
    expect(component.find('Logo')).toHaveLength(1);
    });
    it('Should have HeaderControlPanel', () => {
    const component = mount(<HeaderControlPanel />);
    expect(component.find('HeaderControlPanel')).toHaveLength(1);
    });
});
