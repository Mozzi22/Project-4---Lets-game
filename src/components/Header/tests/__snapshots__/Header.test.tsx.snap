// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should match snapshot 1`] = `"<div class=\\"styled__StHeader-sc-1okobp8-0 kQvxmw\\"><div class=\\"styled__StLogo-sc-1575ivn-0 dODDyP\\"><img src=\\"/public/assets/images/game-logo.svg\\"/><button type=\\"button\\" font-size=\\"18px\\" width=\\"110px\\" class=\\"styled__StButton-sc-15atqtm-0 kNnSXl\\">main_page</button><button type=\\"button\\" font-size=\\"18px\\" width=\\"110px\\" class=\\"styled__StButton-sc-15atqtm-0 kNnSXl\\">statistics</button></div><div class=\\"styled__StControl-sc-1lhkoni-0 hdTeNn\\"><button type=\\"button\\" font-size=\\"26px\\" height=\\"40px\\" width=\\"40px\\" class=\\"styled__StButton-sc-15atqtm-0 kBIpFt\\"></button><select width=\\"70px\\" class=\\"styled__StSelect-sc-1hj7ohk-0 gkgfCL\\"><option id=\\"1\\" value=\\"en\\" class=\\"styled__StOption-sc-1hj7ohk-1 iWufHR\\">en</option><option id=\\"2\\" value=\\"ru\\" class=\\"styled__StOption-sc-1hj7ohk-1 iWufHR\\">ru</option></select></div></div>"`;
